src/program.ts

```diff
@@ -3588,11 +3616,37 @@ export class Function extends TypedElement {
     return local;
   }
 
+  markClosedLocal(local: Local) {
+    console.log("marking local " + local.name + " as closed in function " + this.name);
+    local.envOffset = this.nextEnvOffset;
+    // TODO: the alignment logic is slightly more complicated then this iirc
+    this.nextEnvOffset += local.type.byteSize;
+  }
+
   /* @override */
   lookup(name: string): Element | null {
     var locals = this.localsByName;
     if (locals.has(name)) return assert(locals.get(name));
-    return this.parent.lookup(name);
+
+
+    var parentResult = this.parent.lookup(name);
+    if (parentResult !== null) {
+     /* We found a local in a parent of a function, mark that it must be enclosed
+     * We can infer that since this is a Local and not a Closed Local that the function
+     * depth must be 1 */
+      if (parentResult.kind == ElementKind.LOCAL) {
+        assert(this.parent.kind == ElementKind.FUNCTION);
+        /* Notify the parent that this local must be included in its env. This will also
+         * populate the Local with information about its storage within the Env */
+        (<Function>this.parent).markClosedLocal(<Local>parentResult);
```

Comment:
If I'm wrong about this assertion and it's possible to have a middle element between two functions, then we'll need to figure out a way to port the `makeClosedLocal` call through another `Element`. Not the end of the world but it's a bunch of added boilerplate.

However, if I'm wrong and there could be a local found in an ancestor of a function which does not belong to a function then this would become more complicated.